docs/DeveloperGuide.adoc

```diff
@@ -347,10 +348,15 @@ _{More to be added}_
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-_{More to be added}_
+.  The system is  not intended to be used with more than 1000 entities because student run hackathons rarely exceed this size and should not seem sluggish if the number of entites is less than 1500.
+.  Project is not intended for use on mobile and only should be used on desktop.
+.  The application is better ran from the command line rather than from the graphical user interface.
+.  The application is meant to run offline. 
+.  The application is largely a personnel/HR manager, and is not expected to do anything more than that (eg hackathon finances etc).
+.  The application is to be used for a single hackathon only and not for multiple hackathons.
+.  The application assumes that the hackathon is a short term affair (no longer than 4 days).   
+.  The application assumes that this is an English medium hackathon and that no non-English names are expected.
 
```

Comment:
The GUI should display the result of commands in an intuitive, organized manner that is readable by the laymen(as part of the organization/ affordability of the application).